tests/plugins_test.go

```diff
@@ -143,9 +143,13 @@ func createBufGenYaml(t *testing.T, basedir string, plugin *plugin.Plugin) error
 	defer func() {
 		require.NoError(t, bufGenYaml.Close())
 	}()
+	opts := plugin.ExternalConfig.Registry.Opts
```

Comment:
Changed the tests to prefer running with registry opts (so we continue to get test coverage there - including on our patches).

Reply:
Not sure I follow this change. Shouldn't we rely on the default plugin behavior?

Reply:
We've always run with `default_opts` - it feels good to run with some sort of options if we specify them in buf.plugin.yaml to get more coverage of additional features.

Alternatively we could have two executions of a plugin - one with opts and one without, but that would be 2x the plugin.sum files we check in to this repo.

Reply:
Based on the plugins we have today, do all of them work out-of-the-box without supplying options?

I guess my wish is _every_ plugin in existence shipped with sane options without requiring mandatory ones. IMO that would be a more accurate reflection of its behavior, esp. for `buf generate` users.

Reply:
All of them work out of the box without options. By including options however, we're getting more coverage (including on things like patched plugins like grpc-go).